### PR TITLE
CLN: trim unreachable indexing code

### PR DESCRIPTION
**Pull request by jbrockmendel**
make _set_value not return anything (the DataFrame docstring in particular is misleading)

### CODE REVIEW

File: pandas/core/series.py [hunk 1]
```diff
@@ -1070,7 +1065,7 @@ def _set_with(self, key, value):
 
     def _set_labels(self, key, value):
         key = com.asarray_tuplesafe(key)
-        indexer = self.index.get_indexer(key)
+        indexer: np.ndarray = self.index.get_indexer(key)
```
Comment (WillAyd) [inline pandas/core/series.py]: Shouldn't need to annotate like this; if anything should set the return of `get_indexer` to np.ndarray

**Review by WillAyd — COMMENTED** [review]
This is all just internal right?

File: pandas/core/series.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/series.py]: yah, im trying to pin down what gets passed to _set_values below.  The check on L1075 is never True, so id like to see if we can rip it out

**Comment by jbrockmendel** [comment]
> This is all just internal right?

Yes

File: pandas/core/series.py [hunk 2]
```diff
@@ -902,7 +902,7 @@ def _get_with(self, key):
                         return self._get_values(key)
                 raise
 
-        if not isinstance(key, (list, np.ndarray, Series, Index)):
+        if not isinstance(key, (list, np.ndarray, ExtensionArray, Series, Index)):
```
Comment (jreback) [inline pandas/core/series.py]: we likely don't have many tests which do this on EAs FYI

File: pandas/tests/frame/indexing/test_indexing.py [hunk 3]
```diff
@@ -1369,29 +1369,28 @@ def test_set_value(self, float_frame):
 
     def test_set_value_resize(self, float_frame):
 
-        res = float_frame._set_value("foobar", "B", 0)
-        assert res is float_frame
-        assert res.index[-1] == "foobar"
-        assert res._get_value("foobar", "B") == 0
+        float_frame._set_value("foobar", "B", 0)
+        assert float_frame.index[-1] == "foobar"
```
Comment (jreback) [inline pandas/tests/frame/indexing/test_indexing.py]: you could assert the return value is None

**Review by jreback — CHANGES_REQUESTED** [review]
small assert; i think you can do it in this PR. ping on green.

File: pandas/tests/series/indexing/test_indexing.py [hunk 4]
```diff
@@ -364,16 +364,14 @@ def test_setitem_dtypes():
...
+    datetime_series._set_value(idx, 0)
     assert datetime_series[idx] == 0
 
     # equiv
     s = string_series.copy()
-    res = s._set_value("foobar", 0)
-    assert res is s
-    assert res.index[-1] == "foobar"
-    assert res["foobar"] == 0
+    s._set_value("foobar", 0)
```
Comment (jreback) [inline pandas/tests/series/indexing/test_indexing.py]: same here, assert return value is None

**Comment by jbrockmendel** [comment]
ping

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks

**Pull request merged**